custom/icds_reports/reports/poshan_progress_dashboard_data.py

```diff
@@ -201,38 +209,20 @@ def get_poshan_progress_dashboard_data(domain, year, month, quarter, data_period
     if aggregation_level == 1:
         value_fields.remove('district_name')
     response = {}
-    queryset = PoshanProgressReportView.objects.filter(**filters).order_by(*order_by)
-    if not include_test:
-        queryset = apply_exclude(domain, queryset)
 
     if step == 'aggregated':
+        value_fields.append('num_launched_states')
         if data_period == 'month':
-            data = queryset.aggregate(
-                num_launched_states=Sum('num_launched_states'),
-                num_launched_districts=Sum('num_launched_districts'),
-                num_launched_blocks=Sum('num_launched_blocks'),
-                num_launched_awcs=Sum('num_launched_awcs'),
-                awc_days_open=Sum('awc_days_open'),
-                expected_visits=Sum('expected_visits'),
-                valid_visits=Sum('valid_visits'),
-                pse_eligible=Sum('pse_eligible'),
-                pse_attended_21_days=Sum('pse_attended_21_days'),
-                wer_eligible=Sum('wer_eligible'),
-                wer_weighed=Sum('wer_weighed'),
-                trimester_3=Sum('trimester_3'),
-                counsel_immediate_bf=Sum('counsel_immediate_bf'),
-                height_eligible=Sum('height_eligible'),
-                height_measured_in_month=Sum('height_measured_in_month'),
-                thr_eligible=Sum('thr_eligible'),
-                thr_rations_21_plus_distributed=Sum('thr_rations_21_plus_distributed'),
-                lunch_eligible=Sum('lunch_eligible'),
-                lunch_count_21_days=Sum('lunch_count_21_days'),
-            )
+            data = fetch_month_data(value_fields, order_by, filters, include_test, domain)
```

Comment:
could you explain how will the result of this fetch_month_data be equal to the aggregate query written previously?

Reply:
is that handled by https://github.com/dimagi/commcare-hq/pull/27989/files#diff-0dc9eef06c1966d55e27f777dc1945fbR58?

Reply:
previously we are aggregating at the query level (but for monthly only) but for quarter we are adding row by row in this I shifted month to row by row addition also (to make it generic) as well the values are all integers so sum could be done anywhere, in-fact the PPR is using this approach.